dev-docs/source/UnitTestGoodPractice.rst

```diff
@@ -0,0 +1,191 @@
+   can be used to check stress and timings.
+-  Untestable code is a code-smell, if you can't get the code under test
+   it probably needs refactoring.
+-  Weight your testing to be destructive rather than demonstrative.
+   Destructive tests have a higher efficacy for finding bugs.
+
+Mantid-specific Guidelines
+##########################
+
+-  As noted above, you can assume that individual tests within a cxxtest
+   suite will be run in order.
+-  There must be **no relative paths** (or, more obviously, absolute
+   ones) used in tests as with CMake the code can be build anywhere with
+   respect to the source tree. Make use of the datasearch.directories
+   property (which CMake configures to hold correct paths for a given
+   build).
+-  Ideally, test suites should not have a constructor. If one is
+   required, the following boiler-plate code **must** be inserted in the
+   test class:
+
+   .. code-block:: c++
+
+      static NameOfTest *createSuite() { return new NameOfTest(); }
+      static void destroySuite(NameOfTest *suite) { delete suite; }
+
+   where ``NameOfTest`` is the name of the test class. Without this, the
+   class is turned into a static meaning that the constructor is run at
+   initialisation even if (via an argument) you are not going to run that
+   particular test suite. Also, this can cause problems if running tests in
+   parallel.
+
+-  Be cautious in use of the setUp() and tearDown() methods. Be aware
+   that if you use these in your suite they will be run before/after
+   **every single** individual test. That's fine if it's the behaviour
+   you really need, but we have found that to be rare - use the
+   constructor or set things up within the test.
+-  To avoid clashes, use unique names for workspaces that will go into
+   the [Analysis Data Service], perhaps by prepending the name of the
+   test suite. Even better, don't put workspaces into the ADS in the
+   first place: for example, an InputWorkspace property can be set via
+   pointer instead of by name.
+-  Clean up the ADS at (or before) the end of the test suite.
+
+Using files in Unit tests
+-------------------------
+
+Files for unit tests bloat our repository and slow down the testing
+process. Therefore unless the prime purpose of the algorithms is to load
+or save a file then you should not use a file in your unit tests.
+
+How do I get a workspace filled with data?
+    Firstly you want to think about how much data you really need, unit
+    tests need to be fast so you don't want too much data.
+    Secondly you should use and extend helper classes (like
+    `1 <https://github.com/mantidproject/mantid/blob/master/Code/Mantid/Framework/TestHelpers/inc/MantidTestHelpers/WorkspaceCreationHelper.h>`__)
+    to provide the workspaces for you. Keep things as generic as you can
+    and it will help you and others for other tests.
+    More details of this will be provided at `Testing Utilities <TestingUtilities>`__.
```

Comment:
This link does not work for me.

Is this a result of the pages not yet being in the doctree or is there a `.html` missing?